python/SECRETSMANAGER_MAX_SECRET_AGE/SECRETSMANAGER_MAX_SECRET_AGE.py

```diff
@@ -0,0 +1,424 @@
+            return 'COMPLIANT'
+        return 'NON_COMPLIANT'
+
+    secret_versions = AWS_SECRETSMANAGER_CLIENT.list_secret_version_ids(
+        SecretId=secret.get('Name'),
+        IncludeDeprecated=False
+    ).get('Versions')
+    for version in secret_versions:
+        if 'AWSCURRENT' in version.get('VersionStages'):
+            if datetime.replace(version.get('CreatedDate'), tzinfo=timezone.utc) > max_secret_age:
+                return 'COMPLIANT'
+    return 'NON_COMPLIANT'
+
+def evaluate_compliance(event, configuration_item, valid_rule_parameters):
+    """Form the evaluation(s) to be return to Config Rules
+
+    Return either:
+    None -- when no result needs to be displayed
+    a string -- either COMPLIANT, NON_COMPLIANT or NOT_APPLICABLE
+    a dictionary -- the evaluation dictionary, usually built by build_evaluation_from_config_item()
+    a list of dictionary -- a list of evaluation dictionary , usually built by build_evaluation()
+
+    Keyword arguments:
+    event -- the event variable given in the lambda handler
+    configuration_item -- the configurationItem dictionary in the invokingEvent
+    valid_rule_parameters -- the output of the evaluate_parameters() representing validated parameters of the Config Rule
+
+    Advanced Notes:
+    1 -- if a resource is deleted and generate a configuration change with ResourceDeleted status, the Boilerplate code will put a NOT_APPLICABLE on this resource automatically.
+    2 -- if a None or a list of dictionary is returned, the old evaluation(s) which are not returned in the new evaluation list are returned as NOT_APPLICABLE by the Boilerplate code
+    3 -- if None or an empty string, list or dict is returned, the Boilerplate code will put a "shadow" evaluation to feedback that the evaluation took place properly
+    """
+
+    ###############################
+    # Add your custom logic here. #
+    ###############################
+
+    evaluations = []
+    paginator = AWS_SECRETSMANAGER_CLIENT.get_paginator('list_secrets')
+
+    for secret_list in paginator.paginate():
+        for secret in secret_list['SecretList']:
+            secret_arn = secret.get('ARN')
+            evaluations.append(build_evaluation(secret_arn, evaluate_secret_compliance(valid_rule_parameters, secret), event, resource_type=DEFAULT_RESOURCE_TYPE, annotation=None))
+
+    return evaluations
+
+def evaluate_parameters(rule_parameters):
+    """Evaluate the rule parameters dictionary validity. Raise a ValueError for invalid parameters.
+
+    Return:
+    anything suitable for the evaluate_compliance()
+
+    Keyword arguments:
+    rule_parameters -- the Key/Value dictionary of the Config Rules parameters
+    """
+    if 'max_secret_age_days' in rule_parameters and int(rule_parameters.get('max_secret_age_days')) > int('999999999'):
+        raise ValueError('max_secret_age_days must be less than 999999999')
+    valid_rule_parameters = rule_parameters
+    return valid_rule_parameters
```

Comment:
Add a global variable on the top `DEFAULT_MAX_SECRET_AGE_DAYS = 30`

then something like this
```
try:
     max_secret_age_days = int(rule_parameters.get('max_secret_age_days', DEFAULT_MAX_SECRET_AGE_DAYS)
except TypeError:
    raise ValueError("...must be an integer")

if max_secret_age_days > 1E9-1:
   raise ValueError("...must be less than ...")

return {"max_secret_age_days" : max_secret_age_days }

Reply:
Added global variable and moved logic to evaluate_parameters()